enterprise/frontend/src/metabase-enterprise/content_translation/utils.ts

```diff
@@ -0,0 +1,38 @@
+import type { DictionaryArray } from "metabase-types/api";
+
+/** Translate a user-generated string
+ *
+ * Terminology: A "msgid" is a 'raw', untranslated string. A "msgstr" is a
+ * translation of a msgid.
+ * */
+export const translateContentString = <
+  MsgidType extends string | null | undefined,
+>(
+  dictionary: DictionaryArray | undefined,
+  locale: string | undefined,
+  /** This argument will be translated only if it is a string. If it is not a
+   * string, it will be returned untranslated. */
+  msgid: MsgidType,
+) => {
+  if (!locale) {
+    return msgid;
+  }
+
+  if (typeof msgid !== "string") {
+    return msgid;
+  }
+
+  if (!msgid.trim()) {
+    return msgid;
+  }
+
+  const msgstr = dictionary?.find(
+    (row) => row.locale === locale && row.msgid === msgid,
+  )?.msgstr;
```

Comment:
Just curious: I noticed that `useListContentTranslationsQuery` already passes in a `locale` so I would expect the `dictionary` to only contain that specific locale. Is this for a scenario where a dictionary contains multiple locale for some reason?